plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy

```diff
@@ -459,23 +462,45 @@ class GoogleBatchTaskHandler extends TaskHandler implements FusionAwareTask {
         final now = System.currentTimeMillis()
         final delta =  now - timestamp;
         if( !taskState || delta >= 1_000) {
-            final status = client.getTaskStatus(jobId, taskId)
-            final newState = status?.state as String
-            if( newState ) {
-                log.trace "[GOOGLE BATCH] Get job=$jobId task=$taskId state=$newState"
-                taskState = newState
-                timestamp = now
-            }
-            if( newState == 'PENDING' ) {
-                final eventsCount = status.getStatusEventsCount()
-                final lastEvent = eventsCount > 0 ? status.getStatusEvents(eventsCount - 1) : null
-                if( lastEvent?.getDescription()?.contains('CODE_GCE_QUOTA_EXCEEDED') )
-                    log.warn1 "Batch job cannot be run: ${lastEvent.getDescription()}"
+            try {
+                final status = client.getTaskStatus(jobId, taskId)
+                inspectTaskStatus(status)
+            }catch (NotFoundException e) {
+                manageNotFound(tasks)
             }
         }
         return taskState
     }
 
+    private void inspectTaskStatus(com.google.cloud.batch.v1.TaskStatus status) {
+        final newState = status?.state as String
+        if (newState) {
+            log.trace "[GOOGLE BATCH] Get job=$jobId task=$taskId state=$newState"
+            taskState = newState
+            timestamp = System.currentTimeMillis()
+        }
+        if (newState == 'PENDING') {
+            final eventsCount = status.getStatusEventsCount()
+            final lastEvent = eventsCount > 0 ? status.getStatusEvents(eventsCount - 1) : null
+            if (lastEvent?.getDescription()?.contains('CODE_GCE_QUOTA_EXCEEDED'))
+                log.warn1 "Batch job cannot be run: ${lastEvent.getDescription()}"
+        }
+    }
+
+    protected String manageNotFound( Iterable<Task> tasks) {
+        // If task is array, check if the in the task list
+        if (tasks.size() > 1) {
+            for (Task t in tasks) {
+                if (t.name == client.generateTaskName(jobId, taskId)) {
+                    inspectTaskStatus(t.status)
+                    return taskState
+                }
+            }
```

Comment:
not sure to understand how status in the case of a array of tasks is handled?

Reply:
There are two ways to check the task status:  the `getTaskStatus` that is failing or looking for the task in the job's task list retrieved at the beginning of the `getTaskState` method. As the first option fails, this is trying with the second option.